### PR TITLE
Document StateQuery

### PR DESCRIPTION
**Pull request by ipdae**
SSIA

### CODE REVIEW

File: NineChronicles.Headless/GraphTypes/StateQuery.cs [hunk 1]
```diff
@@ -27,10 +29,12 @@ public StateQuery()
                 });
             Field<RankingMapStateType>(
                 name: "rankingMap",
+                description: "State for Record AvatarState EXP.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/StateQuery.cs:32]:
```suggestion
                description: "State for AvatarState EXP Record.",
```

File: NineChronicles.Headless/GraphTypes/StateQuery.cs [hunk 2]
```diff
@@ -39,13 +43,16 @@ public StateQuery()
                 });
             Field<ShopStateType>(
                 name: "shop",
+                description: "State for market.",
                 resolve: context => new ShopState((Dictionary) context.Source(Addresses.Shop)));
             Field<WeeklyArenaStateType>(
                 name: "weeklyArena",
+                description: "State for arena.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/StateQuery.cs:50]:
```suggestion
                description: "State for weekly arena.",
```

File: NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs [hunk 3]
```diff
@@ -14,79 +14,103 @@ public AvatarStateType()
         {
             Field<NonNullGraphType<AddressType>>(
                 nameof(AvatarState.address),
+                description: "Address of AvatarState.",
                 resolve: context => context.Source.address);
             Field<NonNullGraphType<IntGraphType>>(
                 nameof(AvatarState.blockIndex),
+                description: "Block index at latest action was executed.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs:21]:
```suggestion
                description: "Block index at the latest executed action.",
```

File: NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs [hunk 4]
```diff
@@ -14,79 +14,103 @@ public AvatarStateType()
...
                 resolve: context => context.Source.blockIndex);
             Field<NonNullGraphType<IntGraphType>>(
                 nameof(AvatarState.characterId),
+                description: "Character ID from CharacterSheet.",
                 resolve: context => context.Source.characterId);
             Field<NonNullGraphType<LongGraphType>>(
                 nameof(AvatarState.dailyRewardReceivedIndex),
+                description: "Block index at DailyReward was executed.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs:29]:
```suggestion
                description: "Block index at the DailyReward execution.",
```

File: NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs [hunk 5]
```diff
@@ -14,79 +14,103 @@ public AvatarStateType()
...
                 nameof(AvatarState.agentAddress),
+                description: "Address of AgentState.",
                 resolve: context => context.Source.agentAddress);
             Field<NonNullGraphType<AddressType>>(
                 nameof(AvatarState.RankingMapAddress),
+                description: "Address of the RankingMapState where this avatar information is recorded.",
                 resolve: context => context.Source.RankingMapAddress);
             Field<NonNullGraphType<LongGraphType>>(
                 nameof(AvatarState.updatedAt),
+                description: "Block index at latest action was executed.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs:41]:
```suggestion
                description: "Block index at the latest executed action.",
```

File: NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs [hunk 6]
```diff
@@ -14,79 +14,103 @@ public AvatarStateType()
...
                 nameof(AvatarState.inventory),
+                description: "AvatarState inventory.",
                 resolve: context => context.Source.inventory);
             Field<NonNullGraphType<ListGraphType<NonNullGraphType<AddressType>>>>(
                 nameof(AvatarState.combinationSlotAddresses),
+                description: "Address list of combination slot.",
                 resolve: context => context.Source.combinationSlotAddresses);
             Field<NonNullGraphType<CollectionMapType>>(
                 nameof(AvatarState.itemMap),
+                description: "List of acquired item ID.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/AvatarStateType.cs:88]:
```suggestion
                description: "List of acquired item ID.",
```

File: NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs [hunk 7]
```diff
@@ -7,10 +7,22 @@ public class InventoryType : ObjectGraphType<Inventory>
     {
         public InventoryType()
         {
-            Field<NonNullGraphType<ListGraphType<NonNullGraphType<ConsumableType>>>>(nameof(Inventory.Consumables));
-            Field<NonNullGraphType<ListGraphType<NonNullGraphType<MaterialType>>>>(nameof(Inventory.Materials));
-            Field<NonNullGraphType<ListGraphType<NonNullGraphType<EquipmentType>>>>(nameof(Inventory.Equipments));
-            Field<NonNullGraphType<ListGraphType<NonNullGraphType<CostumeType>>>>(nameof(Inventory.Costumes));
+            Field<NonNullGraphType<ListGraphType<NonNullGraphType<ConsumableType>>>>(
+                nameof(Inventory.Consumables),
+                description: "List of Consumable."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs:12]:
```suggestion
                description: "List of Consumables."
```

File: NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs [hunk 8]
```diff
@@ -7,10 +7,22 @@ public class InventoryType : ObjectGraphType<Inventory>
...
+            );
+            Field<NonNullGraphType<ListGraphType<NonNullGraphType<MaterialType>>>>(
+                nameof(Inventory.Materials),
+                description: "List of Material."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs:16]:
```suggestion
                description: "List of Materials."
```

File: NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs [hunk 9]
```diff
@@ -7,10 +7,22 @@ public class InventoryType : ObjectGraphType<Inventory>
...
+            );
+            Field<NonNullGraphType<ListGraphType<NonNullGraphType<EquipmentType>>>>(
+                nameof(Inventory.Equipments),
+                description: "List of Equipment."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs:20]:
```suggestion
                description: "List of Equipments."
```

File: NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs [hunk 10]
```diff
@@ -7,10 +7,22 @@ public class InventoryType : ObjectGraphType<Inventory>
...
+            );
+            Field<NonNullGraphType<ListGraphType<NonNullGraphType<CostumeType>>>>(
+                nameof(Inventory.Costumes),
+                description: "List of Costumez."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/Models/Item/InventoryType.cs:24]:
```suggestion
                description: "List of Costumes."
```

File: NineChronicles.Headless/GraphTypes/States/Models/Item/ShopItemType.cs [hunk 11]
```diff
@@ -9,23 +9,29 @@ public ShopItemType()
...
                 nameof(ShopItem.SellerAgentAddress),
+                description: "Address of seller agent.",
                 resolve: context => context.Source.SellerAgentAddress);
             Field<NonNullGraphType<AddressType>>(
                 nameof(ShopItem.SellerAvatarAddress),
+                description: "Address of seller avatar.",
                 resolve: context => context.Source.SellerAvatarAddress);
             Field<NonNullGraphType<GuidGraphType>>(
                 nameof(ShopItem.ProductId),
+                description: "Guid of shop registered.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/Models/Item/ShopItemType.cs:20]:
```suggestion
                description: "Guid of product registered.",
```

File: NineChronicles.Headless/GraphTypes/States/RankingInfoType.cs [hunk 12]
```diff
@@ -9,27 +9,35 @@ public RankingInfoType()
...
                 nameof(RankingInfo.Exp),
+                description: "AvatarState total EXP.",
                 resolve: context => context.Source.Exp);
             Field<NonNullGraphType<IntGraphType>>(
                 nameof(RankingInfo.Level),
+                description: "AvatarState Level.",
                 resolve: context => context.Source.Level);
             Field<NonNullGraphType<IntGraphType>>(
                 nameof(RankingInfo.ArmorId),
+                description: "Equipped Armor id from EquipmentItemSheet.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/RankingInfoType.cs:20]:
```suggestion
                description: "Equipped Armor Id from EquipmentItemSheet.",
```

File: NineChronicles.Headless/GraphTypes/States/RankingInfoType.cs [hunk 13]
```diff
@@ -9,27 +9,35 @@ public RankingInfoType()
...
                 resolve: context => context.Source.ArmorId);
             Field<NonNullGraphType<LongGraphType>>(
                 nameof(RankingInfo.UpdatedAt),
+                description: "RankingInfo updated block index.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/RankingInfoType.cs:24]:
```suggestion
                description: "Block index at RankingInfo update.",
```

File: NineChronicles.Headless/GraphTypes/States/RankingInfoType.cs [hunk 14]
```diff
@@ -9,27 +9,35 @@ public RankingInfoType()
...
                 resolve: context => context.Source.UpdatedAt);
             Field<NonNullGraphType<LongGraphType>>(
                 nameof(RankingInfo.StageClearedBlockIndex),
+                description: "Latest stage cleared block index.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/States/RankingInfoType.cs:28]:
```suggestion
                description: "Block index at latest stage cleared.",
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 15]
```diff
@@ -90,16 +91,17 @@ public ActionMutation()
                 });
 
             Field<NonNullGraphType<TxIdType>>("hackAndSlash",
+                description: "Start stage for get material.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:94]:
```suggestion
                description: "Start stage to get material.",
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 16]
```diff
@@ -173,26 +175,27 @@ public ActionMutation()
                 });
 
             Field<NonNullGraphType<TxIdType>>("combinationEquipment",
+                description: "Combination new equipment.",
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:178]:
```suggestion
                description: "Combine new equipment.",
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 17]
```diff
@@ -173,26 +175,27 @@ public ActionMutation()
...
                     {
                         Name = "recipeId",
-                        Description = "EquipmentRecipe ID."
+                        Description = "EquipmentRecipe ID from EquipmentRecipeSheet."
                     },
                     new QueryArgument<NonNullGraphType<IntGraphType>>
                     {
                         Name = "slotIndex",
-                        Description =  "The index of combination slot. 0 ~ 3"
+                        Description =  "The index of empty combination slot for combination equipment. 0 ~ 3"
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:193]:
```suggestion
                        Description =  "The empty combination slot index to combine equipment. 0 ~ 3"
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 18]
```diff
@@ -228,26 +231,27 @@ public ActionMutation()
...
                     {
                         Name = "itemId",
-                        Description = "Equipment Guid."
+                        Description = "Equipment Guid for upgrade."
                     },
                     new QueryArgument<NonNullGraphType<GuidGraphType>>
                     {
                         Name = "materialId",
-                        Description = "Material Guid."
+                        Description = "Equipment Guid for upgrade material."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:249]:
```suggestion
                        Description = "Material Guid for equipment upgrade."
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 19]
```diff
@@ -228,26 +231,27 @@ public ActionMutation()
...
                     },
                     new QueryArgument<NonNullGraphType<IntGraphType>>
                     {
                         Name = "slotIndex",
-                        Description =  "The index of combination slot. 0 ~ 3"
+                        Description =  "The index of empty combination slot for upgrade equipment. 0 ~ 3"
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:254]:
```suggestion
                        Description =  "The empty combination slot index to upgrade equipment. 0 ~ 3"
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 20]
```diff
@@ -338,21 +343,22 @@ public ActionMutation()
                     }
                 });
             Field<NonNullGraphType<TxIdType>>("sell",
+                description: "Register item to the shop.",
                 arguments: new QueryArguments(
                     new QueryArgument<NonNullGraphType<AddressType>>
                     {
                         Name = "sellerAvatarAddress",
-                        Description = "AvatarState address."
+                        Description = "Avatar address for register shop item."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:351]:
```suggestion
                        Description = "Avatar address to register shop item."
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 21]
```diff
@@ -338,21 +343,22 @@ public ActionMutation()
...
                     },
                     new QueryArgument<NonNullGraphType<GuidGraphType>>
                     {
                         Name = "itemId",
-                        Description = "Item Guid."
+                        Description = "Item Guid to be registered shop."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:356]:
```suggestion
                        Description = "Item Guid to register on shop."
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 22]
```diff
@@ -423,21 +430,22 @@ public ActionMutation()
                 });
 
             Field<NonNullGraphType<TxIdType>>("combinationConsumable",
+                description: "Combination new Consumable.",
                 arguments: new QueryArguments(
                     new QueryArgument<NonNullGraphType<AddressType>>
                     {
                         Name = "avatarAddress",
-                        Description = "AvatarState address."
+                        Description = "Avatar address to combination consumable."
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:438]:
```suggestion
                        Description = "Avatar address to combine consumable."
```

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 23]
```diff
@@ -423,21 +430,22 @@ public ActionMutation()
...
                     {
                         Name = "recipeId",
-                        Description = "ConsumableRecipe ID."
+                        Description = "ConsumableRecipe ID from ConsumableRecipeSheet."
                     },
                     new QueryArgument<NonNullGraphType<IntGraphType>>
                     {
                         Name = "slotIndex",
-                        Description =  "The index of combination slot. 0 ~ 3"
+                        Description =  "The index of empty combination slot for combination consumable. 0 ~ 3"
```
Comment (area363) [inline NineChronicles.Headless/GraphTypes/ActionMutation.cs:448]:
```suggestion
                        Description =  "The empty combination slot index to combine consumable. 0 ~ 3"
```

**Review by area363 — COMMENTED** [review]
🙏

File: NineChronicles.Headless/GraphTypes/ActionMutation.cs [hunk 24]
```diff
@@ -284,26 +288,27 @@ public ActionMutation()
                 });
 
             Field<NonNullGraphType<TxIdType>>("buy",
+                description: "Buy registered shop item.",
                 arguments: new QueryArguments(
                     new QueryArgument<NonNullGraphType<AddressType>>
                     {
                         Name = "sellerAgentAddress",
-                        Description = "ShopItem SellerAgentAddress."
+                        Description = "Agent address from Registered ShopItem."
```
[the remark this answers is not in the document]
Comment (ipdae) [reply NineChronicles.Headless/GraphTypes/ActionMutation.cs:296]: It seems like that by using the IDE's replace function. I will fix it.

**Review by area363 — APPROVED** [review]
👍  Please update branch 🙏

**Review by longfin — APPROVED** [review]

**Pull request merged**